Double maxweight per size above medium

RikerW opened the pull request:
Since weapon weights are doubled, it makes sense you should be able to use appropriately sized items.

RikerW committed:
Double maxweight per size above medium

Since weapon weights are doubled, it makes sense you should be able to use appropriately sized items.

NeroOneTrueKing commented:
I see this doesn't penalize gnomes 😄

NeroOneTrueKing reviewed (changes requested):
There are two places in weapon.c that calculate maxweight of offhand weapons -- one for damage, and one for accuracy. The offhand-weight calculation could be moved into its own function, or you can just do the modification in both places.

Gnomes notwithstanding (it's fine by me if they don't get additional penalties, I suppose).

NeroOneTrueKing committed:
Gnomes get +2 to-hit
A buff!
This makes their early-game accuracy better and should help them use big-people weapons more effectively.

NeroOneTrueKing committed:
Update object colour when polymorphing it.

Commit:
Drow Binders start with empty signet rings

RikerW committed:
Add check in the other function too

RojjaCebolla commented:
just FYI gnomes are still bad

NeroOneTrueKing reviewed (changes requested):
Just noticed another issue, actually.
This commit changes offhand limits like so:
Medium = 1x
Large = 2x
Huge = 4x
Gigantic = 10x

But the weight of those items is 
Medium = 1x
Large = 2x
Huge = 3x
Gigantic = 6x

The formula:
`maxweight *= 2*(youracedata->msize - MZ_MEDIUM);`
should instead be
`maxweight *= 1+(youracedata->msize - MZ_MEDIUM);`

I'm going to ignore the unrelated commits that snuck in; hopefully they won't cause any merge conflicts.
